app/assets/javascripts/discourse/app/components/d-navigation-item.hbs

```diff
@@ -4,7 +4,13 @@
   class={{@class}}
   ...attributes
 >
-  <LinkTo @route={{@route}}>
-    {{yield}}
-  </LinkTo>
+  {{#if @model}}
+    <LinkTo @route={{@route}} @model={{@model}}>
+      {{yield}}
+    </LinkTo>
+  {{else}}
+    <LinkTo @route={{@route}}>
+      {{yield}}
+    </LinkTo>
+  {{/if}}
```

Comment:
What about using @models instead?

Reply:
I think you could even replace all of this by:

```
<LinkTo @route={{@route}} @models={{@models}}>
  {{yield}}
</LinkTo>
```

it will work for `@models === null`, `@models === [1]`, `@model === ["foo", "bar"]`

Reply:
Hmmm you might still need the {{if}} for the # indeed. In any case models seems more future proof than model

Reply:
Actually, we should support both:

https://github.com/emberjs/ember.js/blob/2aba2dedb265231a425189236aea5e7f0e05d2fd/packages/%40ember/-internals/glimmer/lib/components/link-to.ts#L440-L443

@models needs to be an array. Ember overcomplicates things that should be simple.

Reply:
Yes sure there's @model too, it's just you can do `@models=[1]` just like `@model=1`, but technically yes it's probably more correct to have both.

Reply:
I've added support to @models also @jjaffeux. What do you think?